src/plugins/intel_gna/src/layers/gna_convolution_layer.cpp

```diff
@@ -17,12 +17,22 @@
 namespace GNAPluginNS {
 namespace GNAConvolutionLayer {
 
+bool should_transpose_h_w(const uint32_t in_height,
+                          const uint32_t kernel_height,
+                          const uint32_t in_channels,
+                          const uint32_t stride_height) {
+    return in_height == kernel_height && in_channels == 1 && stride_height == 1;
+}
+
 bool isMappableFrom2DTo1D(const uint32_t inHeight, const uint32_t inWidth, const uint32_t in_channels,
                           const uint32_t kernelHeight, const uint32_t kernelWidth,
                           const uint32_t strideHeight, const uint32_t strideWidth) {
-    return ((inHeight > 1 && inWidth > 1) &&
-            ((inWidth == kernelWidth && strideWidth == 1) ||
-             (inHeight == kernelHeight && strideHeight == 1 && in_channels == 1)));
+    if (inHeight <= 1 || inWidth <= 1) {
```

Comment:
I think we don't want to pass h or w == 0 here silently, do we?

Reply:
I don't think so.
It is not directly related to this PR to judge whether h or w == 0 or not, so I did not want to change the logic in this place.

Reply:
This condition looks strange for me too. 
```suggestion
    if (inHeight == 1 || inWidth == 1) {
```

Reply:
Will left as is for now (not want to change logic in CF/FF), but we can get back after 2022.3